src/graphforge/datasets/formats/json_graph.py

```
@@ -0,0 +1,353 @@
+
+        # distance: value must be non-negative number
+        elif t == "distance":
+            if not isinstance(v, (int, float)) or isinstance(v, bool):
+                raise ValueError(f"Type 'distance' requires numeric value, got {type(v).__name__}")
+            if v < 0:
+                raise ValueError(f"Type 'distance' requires non-negative value, got {v}")
+
+        # list: value must be array of PropertyValue objects
+        elif t == "list":
+            if not isinstance(v, list):
+                raise ValueError(f"Type 'list' requires array value, got {type(v).__name__}")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)